include/xbps.h.in

```diff
@@ -1921,6 +1921,42 @@ char *xbps_xasprintf(const char *fmt, ...)__attribute__ ((format (printf, 1, 2))
  */
 bool xbps_mmap_file(const char *file, void **mmf, size_t *mmflen, size_t *filelen);
 
+/**
+ * Compares the mode of the file \a file with the mode
+ * value specified by \a mode.
+ *
+ * @param[in] file Path to a file.
+ * @param[in] mode Mode value to compare.
+ *
+ * @return 0 if \a file and \a mode have the same value, ERANGE
+ * if it differs, or any other errno value on error.
+ */
+int xbps_file_mode_check(const char *file, const mode_t mode);
+
+/**
+ * Compares the owner of the file \a file with the owner
+ * name specified by \a owner.
+ *
+ * @param[in] file Path to a file.
+ * @param[in] owner Owner name to compare.
+ *
+ * @return 0 if \a file and \a owner have the same value, ERANGE
+ * if it differs, or any other errno value on error.
+ */
+int xbps_file_owner_check(const char *file, const char *owner);
+
+/**
+ * Compares the group of the file \a file with the group
+ * name specified by \a group.
+ *
+ * @param[in] file Path to a file.
+ * @param[in] group Group name to compare.
+ *
+ * @return 0 if \a file and \a group have the same value, ERANGE
+ * if it differs, or any other errno value on error.
+ */
+int xbps_file_group_check(const char *file, const char *group);
+
```

Comment:
Sorry, I misunderstood the question about putting file checks into libxbps, I don't think those functions are relevant/provide enough benefit to add them to the public api.